widget/fileicon.go

```diff
@@ -82,9 +78,11 @@ func (i *FileIcon) CreateRenderer() fyne.WidgetRenderer {
 	i.propertyLock.RLock()
 	defer i.propertyLock.RUnlock()
 
-	i.updateURI(i.URI)
+	i.setURI(i.URI)
+
 	s := &fileIconRenderer{file: i}
 	s.updateObjects()
+	i.cachedURI = i.URI
```

Comment:
isn't setURI above going to do this?

Reply:
I don't think it should. If it did, the updating of new objects in refresh would not be called otherwise.
Refresh alls updates the cachedURI after objects have been updated. A bit like how the icon caching for widget.Icon sets cache to `nil` and then updates after items have been updated.

Reply:
But thinking about this, it might no be needed here specifically. Because the widget will be calling `Refresh()` before it is shown, right?

Reply:
Apolgies, I misread line 65